### PR TITLE
added dav media provider

### PR DESCRIPTION
**Pull request by joicemjoseph**

### CODE REVIEW

File: cmd/settings.go [hunk 1]
```diff
@@ -32,17 +32,22 @@ type settings struct {
...
+	UploadS3Bucket             string            `json:"upload.s3.bucket"`
+	UploadS3BucketDomain       string            `json:"upload.s3.bucket_domain"`
+	UploadS3BucketPath         string            `json:"upload.s3.bucket_path"`
+	UploadS3BucketType         string            `json:"upload.s3.bucket_type"`
+	UploadS3Expiry             string            `json:"upload.s3.expiry"`
+	UploadDavEndpoint          string            `json:"upload.webdav.endpoint"`
+	UploadDavUsername          string            `koanf:"upload.webdav.username"`
+	UploadDavPassword          string            `koanf:"upload.webdav.password"`
+	UploadDavRootPath          string            `koanf:"upload.webdav.root_path"`
+	UploadDavHeaders           map[string]string `koanf:"upload.webdav.headers"`
```
Comment (joicemjoseph) [inline cmd/settings.go:51]: Should we add 

```if set.UploadDavEndpoint != "" 
	if !dav.Connect(dav.Opts{
		Endpoint: set.UploadDavEndpoint,
		Username: set.UploadDavUsername,
		Password: set.UploadDavPassword,
		RootPath: set.UploadDavRootPath,
		Headers:  set.UploadDavHeaders,
	}) {
		return echo.NewHTTPError(http.StatusBadRequest,
			fmt.Sprintf("Error connecting to webdav provider: %v", err))
	}
```

in line number 187 as an additional validation for configurations

**Comment by joicemjoseph** [comment]
image

**Comment by knadh** [comment]
Can you re-base and refactor this PR?

**Comment by joicemjoseph** [comment]
sure!

**Comment by joicemjoseph** [comment]
Rebased.

image

**Comment by knadh** [comment]
Closing this for now since there's no clarity on how WebDav can function without local disk support anyway, in which case, there doesn't seem to be a point support WebDav to begin with.

**Pull request closed without merging**